doc/nrf/releases/release-notes-changelog.rst

```diff
@@ -104,6 +104,18 @@ Bluetooth libraries and services
 
 * |no_changes_yet_note|
 
+Common Application Framework (CAF)
+----------------------------------
+
+* Added:
+
+  * :c:func:`module_name_get` function has been added.
+
+* Updated:
+
+  * Implementation for getting module id is fixed.
+    Until now ``MODULE_ID`` macro and :c:func:`module_id_get` function has incorrectly returned module name instead of module id.
```

Comment:
> function has incorrectly returned module name instead of module id

I wouldn't say this was incorrect. The original implementation used only strings - we used the fact that the same string ends up being in one location after linking. To maintain the compatibility Radek left the module id reference to the string while introducing the module indexes. We probably should change it at that time. I don't recall now why we did not - it could be me that was against it or we thought that it's better to leave module reference as is. Anyway it seems now we will have more and more problems if we use two reference locations - one in the array and one being a string.

Let's mention that in release notes in something like:
> Unify module id reference location. The array holding module reference objects is explicitly defined in linker script to avoid creating an orphan section. The module name cannot be obtained from reference object directly but a helper function should be used instead.

Reply:
Second sentence is about other PR that is already merged #6427, but i got your point. I will put something like:
```
Unify module id reference location. `MODULE_ID`` macro and :c:func:`module_id_get` function
now returns module reference from dedicated section of instead of module name.
The module name can not be obtained from reference object directly, a helper function
(`module_name_get`) should be used instead.

Reply:
Actually i could add this sentence too. Nvm, will do.

Reply:
It doesn't really matter if it's about other PR. Since we have not added it there we can add it now. This things are connected.

Reply:
Yes, exactly. I didn't think about this that way when writing first comment. Updated as requested.